torch/__init__.py

```diff
@@ -47,7 +48,7 @@
     'are_deterministic_algorithms_enabled',
     'is_deterministic_algorithms_warn_only_enabled',
     'set_deterministic_debug_mode', 'get_deterministic_debug_mode',
-    'set_warn_always', 'is_warn_always_enabled',
+    'set_warn_always', 'is_warn_always_enabled', 'extend_library'
```

Comment:
A thought here: should we be putting this function in the top-level torch namespace? An alternative is to place it in `torch.ops` (hazard: `extend_library` would then no longer be a valid namespace--though it's hard to imagine anyone wanting to write a namespace this way), `torch.overrides` (this is not quite right, as overrides is about `torch_function` level override and that's not what you're doing here) or `torch.library` (this is symmetric with the C++ header `torch/library.h`).

I think we should seriously consider putting this in `torch.library`

Reply:
> torch.library (this is symmetric with the C++ header torch/library.h)

That's a very reasonable suggestion and makes a lot of sense. In general, what are some general rules to assess if something should be added to the `torch` namespace? For example, how do we decide if we should only expose it in `torch.library` or expose it in both `torch` and `torch.library`?